src/neo4j_genai/components/entity_relation_extractor.py

```diff
@@ -0,0 +1,114 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+import enum
+import json
+import logging
+from typing import Any, Union
+
+from pydantic import ValidationError, validate_call
+
+from neo4j_genai.components.schema import SchemaConfig
+from neo4j_genai.components.types import Neo4jGraph, TextChunks
+from neo4j_genai.exceptions import LLMGenerationError
+from neo4j_genai.generation.prompts import ERExtractionTemplate, PromptTemplate
+from neo4j_genai.llm import LLMInterface
+from neo4j_genai.pipeline.component import Component
+
+logger = logging.getLogger(__name__)
+
+
+class EntityRelationExtractor(Component):
+    async def run(self, chunks: TextChunks, **kwargs: Any) -> Neo4jGraph:
+        return Neo4jGraph(nodes=[], relationships=[])
+
+
+class OnError(enum.Enum):
+    RAISE = "RAISE"
+    IGNORE = "CONTINUE"
+
+
+class LLMEntityRelationExtractor(EntityRelationExtractor):
+    def __init__(
+        self,
+        llm: LLMInterface,
+        prompt_template: ERExtractionTemplate | str = ERExtractionTemplate(),
+        on_error: OnError = OnError.RAISE,
+    ) -> None:
+        self.llm = llm  # with response_format={ "type": "json_object" },
+        if isinstance(prompt_template, str):
+            template = PromptTemplate(prompt_template, expected_inputs=[])
+        else:
+            template = prompt_template
+        self.prompt_template = template
+        self.on_error = on_error
+
+    def update_ids(self, chunk_index: int, graph: Neo4jGraph) -> Neo4jGraph:
+        for node in graph.nodes:
+            node.id = f"{chunk_index}:{node.id}"
```

Comment:
Is this going to be sufficient as a unique ID? If the user runs an extractor several times and uploads to the same graph having IDs like '1:1' might cause issues. Maybe we could include something like a timestamp to ensure its always unique across multiple runs

Reply:
Good point! https://github.com/neo4j/neo4j-genai-python/pull/85/commits/9ea75d993f2e827bef5a216517202157c3c2480c